### PR TITLE
Email worker

### PR DESCRIPTION
**Pull request by Josephu**
這個worker會根據設定的時間寄信。

目前主要的設定在default.yaml
enabled - 可否寄信。唯有在環境成熟後才會turn on，default off，所以目前是不會寄信的
schedule - '00 * * * * *' it will send every minute, so this is only for testing use.

其他主要的change都是因為資料結構更新，所以code要跟著改。

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,6 @@ Once MySql server is installed and the root user is configured, execute the foll
 ```
 yarn db-create  // create database
 yarn db-migrate // migrate database
-yarn db-seed    // seed database
```
Comment (Josephu) [inline README.md]: seed is not used anymore. Clean up readme.

File: api/service/send-email-service.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const { EmailListItem } = require('../models/email-list-item');
 
 // This is mocked out for now, will be complete when DB is done
 function getEmailList() {
-  const emailCsvFilePath = 'db/data/email-list.csv';
+  const emailCsvFilePath = config.get('reminderEmail.emailListFilePath');
```
Comment (Josephu) [inline api/service/send-email-service.js]: we need different file name for local testing and server, so I have made it configurable.

File: api/service/send-email-service.js [hunk 3]
```diff
@@ -67,10 +67,13 @@ async function buildEventsForMultipleServices(from, to) {
         { from, to },
         service.name
       );
-      events[service.name] = EventMapper.convertEventsModelToDto(
+      events[service.name] = EventMapper.groupEventsByCalendarDate(
```
Comment (Josephu) [inline api/service/send-email-service.js]: @yhua045 has renamed the method names, so this has to be renamed too.

File: api/service/send-email-service.js [hunk 4]
```diff
@@ -67,10 +67,13 @@ async function buildEventsForMultipleServices(from, to) {
...
         eventsForService
       );
-      events[service.name].lang = service.locale;
+      events[service.name] = events[service.name].map((event) => {
```
Comment (Josephu) [inline api/service/send-email-service.js]: My code was wrong before, fixing it up.

File: scripts/email/test/generate.js [hunk 5]
```diff
@@ -8,71 +8,71 @@ const mockData = {
...
-        { role: '招待', name: '吳愛倫' },
-        { role: '司獻', name: '傅慧' },
-        { role: '司獻', name: '何平' },
-        { role: '聖餐<br/>聖洗', name: '簡瑞蘭' },
-        { role: '投影', name: '宣志凌' },
-        { role: '音控', name: '紀哲威' },
-        { role: '燈光', name: '許天因' },
-        { role: '督堂', name: '許天因' },
-        { role: '愛餐', name: '平安' }
+      positions: [
```
Comment (Josephu) [inline scripts/email/test/generate.js]: The naming is changed due to changes from @yhua045 's previous PR

File: test/unit-test/send-email-service-test.js [hunk 6]
```diff
@@ -6,13 +6,13 @@ const { EmailListItem } = require('../../api/models/email-list-item');
 describe('Send Email Service - Parsing Email CSV File and Build Email String', function() {
   this.timeout(5000);
   it('returns only items that have either a English or Chinese name or both and a valid email address or empty email', function() {
-    const emailCsvFilePath = 'db/data/email-list.csv';
+    const emailCsvFilePath = 'db/data/email-list-example.csv';
```
Comment (Josephu) [inline test/unit-test/send-email-service-test.js]: I used `email-list-example.csv` instead for local so I can use `email-list.csv` for qa and production.

**Comment by Josephu** [comment]
@josephj  目前只要server跑 job就會跑。先看gmail如何。關於email 平台你有什麼想法？

**Comment by josephj** [comment]
@Josephu 只要我能收到就可以在不同平台測試，可能 gmail 收到後就轉寄到 Outlook, Yahoo 來看看囉

**Pull request merged**